backend/bitboxbase/bitboxbase.go

```diff
@@ -228,44 +263,220 @@ func (base *BitBoxBase) RPCClient() *rpcclient.RPCClient {
 // MiddlewareInfo returns the received MiddlewareInfo packet from the rpcClient
 func (base *BitBoxBase) MiddlewareInfo() (rpcmessages.SampleInfoResponse, error) {
 	if !base.active {
-		err := errp.New("Attempted a call to non-active base")
-		return rpcmessages.SampleInfoResponse{}, err
+		return rpcmessages.SampleInfoResponse{}, errp.New("Attempted a call to non-active base")
 	}
 	return base.rpcClient.GetSampleInfo()
 }
 
 // VerificationProgress returns the received VerificationProgress packet from the rpcClient
 func (base *BitBoxBase) VerificationProgress() (rpcmessages.VerificationProgressResponse, error) {
 	if !base.active {
-		err := errp.New("Attempted a call to non-active base")
-		return rpcmessages.VerificationProgressResponse{}, err
+		return rpcmessages.VerificationProgressResponse{}, errp.New("Attempted a call to non-active base")
 	}
 	return base.rpcClient.GetVerificationProgress()
 }
 
-// SyncWithOption returns true if the chosen sync option was executed successfully
-func (base *BitBoxBase) SyncWithOption(option SyncOption) (bool, error) {
+// ReindexBitcoin returns true if the chosen sync option was executed successfully
+func (base *BitBoxBase) ReindexBitcoin() error {
+	if !base.active {
+		return errp.New("Attempted a call to non-active base")
+	}
+	base.log.Println("bitboxbase is making a ReindexBitcoin call")
+	reply, err := base.rpcClient.ReindexBitcoin()
+	base.changeStatus(bitboxbasestatus.StatusInitialized)
+	if err != nil {
+		base.log.WithError(err)
+		// if the rpc client throws an error we don't want to pass that on to the frontend, because its probably junk
```

Comment:
The frontend needs to know there was an error anyway, and not pretend it worked.

It is correct to return an error and let the handler or the frontend deal with it.